modules/external-api/src/main/java/org/opencastproject/external/util/SchedulingUtils.java

```diff
@@ -0,0 +1,395 @@
+   *          The conflicting {@link MediaPackage}s.
+   * @param indexService
+   *          The {@link IndexService} for getting the corresponding events for the conflicting {@link MediaPackage}s.
+   * @param externalIndex
+   *          The ExternalIndex to use for getting the corresponding events for the conflicting MediaPackages.
+   *
+   * @return A List of conflicting events, represented as JSON objects.
+   *
+   * @throws SearchIndexException
+   *          If an event cannot be found.
+   */
+  public static List<JValue> convertConflictingEvents(
+      Optional<String> checkedEventId,
+      List<MediaPackage> mediaPackages,
+      IndexService indexService,
+      AbstractSearchIndex externalIndex
+  ) throws SearchIndexException {
+    final List<JValue> result = new ArrayList<>();
+    for (MediaPackage mediaPackage : mediaPackages) {
+      final Opt<Event> eventOpt = indexService.getEvent(mediaPackage.getIdentifier().compact(), externalIndex);
+      if (eventOpt.isSome()) {
+        final Event event = eventOpt.get();
+        if (checkedEventId.isPresent() && checkedEventId.equals(event.getIdentifier())) {
+          continue;
+        }
+        result.add(obj(f("start", v(event.getTechnicalStartTime())), f("end", v(event.getTechnicalEndTime())),
+            f("title", v(event.getTitle()))));
+      } else {
+        logger.warn("Index out of sync! Conflicting event catalog {} not found on event index!",
+            mediaPackage.getIdentifier().compact());
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Get the conflicting events for the given SchedulingInfo.
+   *
+   * @param schedulingInfo
+   *          The SchedulingInfo to check for conflicts.
+   * @param agentStateService
+   *          The {@link CaptureAgentStateService} to use for retrieving capture agents.
+   * @param schedulerService
+   *          The {@link SchedulerService} to use for conflict checking.
+   * @return
+   *          A list of {@link MediaPackage} elements which cause conflicts with the given SchedulingInfo.
+   *
+   * @throws NotFoundException
+   *          If the capture agent cannot be found.
+   * @throws UnauthorizedException
+   *          If the {@link SchedulerService} cannot be queried due to missing authorization.
+   * @throws SchedulerException
+   *          In case internal errors occur within the {@link SchedulerService}.
+   */
+  public static List<MediaPackage> getConflictingEvents(
+      SchedulingInfo schedulingInfo,
+      CaptureAgentStateService agentStateService,
+      SchedulerService schedulerService
+  ) throws NotFoundException, UnauthorizedException, SchedulerException {
+
+    if (schedulingInfo.getRrule().isSome()) {
+      final Agent agent = agentStateService.getAgent(schedulingInfo.getAgentId().get());
+      String timezone = agent.getConfiguration().getProperty("capture.device.timezone");
+      if (StringUtils.isBlank(timezone)) {
+        timezone = TimeZone.getDefault().getID();
```

Comment:
This code is taken from the SchedulerService which turns out to be buggy. When the SchedulerService is fixed, we should update this code as well.